ses-lambda-sam/README.md

```diff
@@ -0,0 +1,99 @@
+
+## Deployment Instructions
+
+1. Create a new directory, navigate to that directory in a terminal and clone the GitHub repository:
+   ```bash
+   git clone https://github.com/aws-samples/serverless-patterns
+   ```
+2. Change directory to the pattern directory:
+   ```bash
+   cd ses-lambda-sam
+   ```
+3. From the command line, use AWS SAM to deploy the AWS resources for the pattern as specified in the template.yml file:
+   ```bash
+   sam deploy -g
+   ```
+4. During the prompts:
+
+   - Enter a stack name. 
+   - Enter the desired AWS Region. Please refer to the [documentation](https://docs.aws.amazon.com/ses/latest/dg/regions.html#region-receive-email) for the list of supported regions.
+   - Enter `DomainName`. This is your Route 53 public hosted zone name. Please refer to the prerequisite section for additional details.
+   - Enter `ResourceRecords`. Depending on your selected region, please get the Email Receiving Endpoint from this [AWS Documentation](https://docs.aws.amazon.com/ses/latest/dg/regions.html#region-receive-email). Then you need to provide the value as `10 {Email Receiving Endpoint}`. For example, for US East N. Virginia region, please enter `10 inbound-smtp.us-east-1.amazonaws.com`. 
+   - Leave rest of the options as default.
+
+5. Note the outputs from the SAM deployment process. This contain the resource names and/or Ids which are used for next step as well as for testing.
+
+6. From the command line, please execute the below command. Please replace `EmailReceiptRuleSet` from the `sam deploy` output and also your region. This will activate the Amazon SES ruleset to invoke the AWS Lambda function once it receives the email.
+   ```bash
+   aws ses set-active-receipt-rule-set  --rule-set-name {EmailReceiptRuleSet} --region {your-region}
+   ```
+
+7. By now, you should receive an email confirmation with subject line `DKIM setup SUCCESS for {DomainName} in {your-region}`. 
+
+
+
+
+## How it works
+
+
+Please refer to the architecture diagram below:
+
+![End to End Architecture](images/architecture.png)
+
+* User sends email to the configured domain in Amazon SES.
+* Amazon SES receives the email and then applies the rule set to trigger an AWS Lambda function. 
+* The AWS Lambda functoin processes information from the SES event.
+
+## Testing
+
+
+1. From the command line, please execute the below command to tail logs of the AWS Lambda function. Please replace `FunctionName` from the `sam deploy` output and also your region. 
+   ```bash
+   aws logs tail --follow /aws/lambda/{FunctionName} --region {your-region}
+   ```
+
+
+2. Make a note of `ReceivingEmailAddress` from the `sam deploy` output and send email to that email id. Send an email to this Id.
+
+
+3. You should see logline similar to following in the Lambda function log: 
```

Comment:
Is this in near real-time or should the user expect some delay?

Reply:
It is event-driven. For me it took <5 sec. So I have added line to expect a few seconds of delay.